ui/lib/pki/addon/components/page/roles/role/details-page.ts

```diff
@@ -0,0 +1,35 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+
+// interface Attribute {
+//   name: string;
+//   options?: {
+//     label?: string;
+//   };
+// }
+
+// TODO: pull this in from route model once it's TS
+interface Args {
+  role: {
+    backend: string;
+    id: string;
+  };
+}
+
+export default class DetailsPage extends Component<Args> {
+  get breadcrumbs() {
+    return [
+      { label: this.args.role.backend || 'pki', path: 'overview' },
+      { label: 'roles', path: 'roles.index' },
+      { label: this.args.role.id },
+    ];
+  }
+
+  get arrayAttrs() {
+    return ['keyUsage', 'extKeyUsage', 'extKeyUsageOids'];
+  }
```

Comment:
Should these have type definitions for what they return or is that something that will be added later?

Reply:
Since it's returning a static value, the type is inferred. So if we use `this.arrayAttrs` somewhere else in this class, it will already know it's type `string[]`. Unfortunately we don't have type-checking for templates yet so it's a little moot but nice that Typescript infers where it can!